airflow/providers/ssh/hooks/ssh.py

```diff
@@ -242,14 +242,19 @@ def __init__(
                 ssh_conf.parse(config_fd)
             host_info = ssh_conf.lookup(self.remote_host)
             if host_info and host_info.get('proxycommand'):
-                self.host_proxy = paramiko.ProxyCommand(host_info['proxycommand'])
+                self.host_proxy_cmd = host_info['proxycommand']
 
             if not (self.password or self.key_file):
                 if host_info and host_info.get('identityfile'):
                     self.key_file = host_info['identityfile'][0]
 
         self.port = self.port or SSH_PORT
 
+    @property
```

Comment:
Can probably make this a cached_property too:

```suggestion
    @cached_property
```

with following imports:

```python
try:
    from functools import cached_property
except ImportError:
    from cached_property import cached_property
```

Reply:
I thought about this but wasn't sure how the hook would be used.  Especially not sure if a proxy (which is actually an SSH session started at its `__init__`) would become inactive and not usable if we keep it long enough in the cache.  But think about the current behavior (of creating the proxy and keeping it during the life of `SSHHook`), it is actually the closest.

So I'll make the changes as suggested.  Thank you.